languages/go/oso.go

```diff
@@ -50,17 +50,19 @@ func (o Oso) ClearRules() error {
 
 /*
 Register a Go type so that it can be referenced within Polar files.
+Accepts a constructor function, or nil if no constructor is required.
 */
-func (o Oso) RegisterClass(cls reflect.Type) error {
-	return (*o.p).registerClass(cls, nil)
+func (o Oso) RegisterClass(cls reflect.Type, cptr interface{}) error {
```

Comment:
Is there a more specific type we can use for the constructor? `interface{}` accepts anything, right?

Something like a function pointer?

Reply:
Not really, there's no general function pointer type, there's just specific function types like `func (int, int) int`. We don't know any of those things since they're different for every constructor.